File: Web design and  UI technologies/JavaScript OOP/04 TypeScript/TypeScriptHW/scripts/malls/Person.ts
module Malls {
    export class Person implements IPerson {
        constructor(public firstName: string, public lastName: string, public sex: Sex, public age?: number) {
        }

        toString(): string {
            var age = this.age || 'not available';
            var sex = this.sex === 0 ? 'male' : 'female';
            var result = 'Person info:\n' + this.firstName + ' ' + this.lastName + '\nage: ' + age + '\nsex: ' + sex;
            return result;
        }
    }
} 

File: Programming with C#/C# OOP/06 Common-Type-System/04. Person task/Person.cs
namespace _04.Person_task
{
    public class Person
    {
        private string name;
        private int? age;

        public string Name
        {
            get { return this.name; }
            set { this.name = value; }
        }

        public int? Age
        {
            get { return this.age; }
            set { this.age = value; }
        }

        public override string ToString()
        {
            string result = null;

            if (this.Age.HasValue)
            {
                result = this.Name + ", age: " + this.Age;
            }
            else
            {
                result = this.Name + ", age: unavailable";
            }

            return result;
        }
    }
}
